### ISSUE TITLE
[BUG] Support casting with Pandas dtype aliases

### ISSUE THREAD

**Issue by sarahyurick** [rapidsai/cudf#8904]
Related to #8885. 

I am able to cast to Pandas' boolean dtype with: 
```
import cudf
import pandas as pd

s = cudf.Series([1, 2, 0])
s.astype(pd.BooleanDtype())
0     True
1     True
2    False
dtype: bool
```

However, the string name "boolean" for the BooleanDtype is not supported: 
```
s.astype("boolean")
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/tmp/ipykernel_26633/182000827.py in <module>
      4 s = cudf.Series([1, 2, 3])
      5 s.astype(pd.BooleanDtype())
----> 6 s.astype("boolean")

~/miniconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/series.py in astype(self, dtype, copy, errors)
   3114         except Exception as e:
   3115             if errors == "raise":
-> 3116                 raise e
   3117             elif errors == "warn":
   3118                 import traceback

~/miniconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/series.py in astype(self, dtype, copy, errors)
   3106         try:
   3107             print(dtype)
-> 3108             data = self._column.astype(dtype)
   3109 
   3110             return self._copy_construct(

~/miniconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/column/column.py in astype(self, dtype, **kwargs)
    884         dtype = pandas_dtypes_to_cudf_dtypes.get(dtype, dtype)
    885         if _is_non_decimal_numeric_dtype(dtype):
--> 886             return self.as_numerical_column(dtype, **kwargs)
    887         elif is_categorical_dtype(dtype):
    888             return self.as_categorical_column(dtype, **kwargs)

~/miniconda3/envs/cudf_dev/lib/python3.8/site-packages/cudf/core/column/numerical.py in as_numerical_column(self, dtype, **kwargs)
    254 
    255     def as_numerical_column(self, dtype: Dtype, **kwargs) -> NumericalColumn:
--> 256         dtype = np.dtype(dtype)
    257         if dtype == self.dtype:
    258             return self

TypeError: data type 'boolean' not understood
```

**Comment by sarahyurick** [comment]
Closing, as this is included in #8915.

**Comment by galipremsagar** [comment]
Fixed by #8920
